src/utils/electron/openInVsCode.ts

```diff
@@ -1,15 +1,29 @@
 import type { STATE } from 'src/state/state'
-import { exec } from 'child_process'
+import { exec, execSync } from 'child_process'
 import { existsSync } from 'fs'
 import { resolve as pathResolve } from 'pathe'
 
 import { cwd } from 'process'
-import { toastError } from '../misc/toasts'
+import { toastError, toastInfo } from '../misc/toasts'
 
 const workspaceFolderPath = cwd()
 
-export async function openInVSCode(st: STATE, filePathWithinWorkspace: string): Promise<void> {
+/* Probably un-needed function, but this makes sure that we keep the UI non-blocking */
+async function tryEditor(editor: string, filePath: string): Promise<void> {
     return new Promise((resolvePromise, rejectPromise) => {
+        exec(`${editor} ${filePath}`, (error, stdout, stderr) => {
```

Comment:
It will block anyway / will not change anything. The `new Promise()` construction starts the promise right away, and since the lambda never yields control (no await; await are when the green thread will yield control), promise will finish synchronously I think. I'll check my mental model, but strong confidence.




I think we just need to not sure exec but some async alternative.
I think Exec is even massively slow because it uses the shell if i'm correct, will also have to check that. We probably want to spawn instead using.tmvscode binary path directly, possibly with some code or mib to detect.it's location in a cross platform way






But yeah, my usage of sync IO operations need to be adressed at some point; most sync operations need to be made async. I did that early to keep things simple and debug more efficient.

Reply:
Will fix typos in my messages when in front of a keyboard